### PR TITLE
Move DevMenu to XAML markup

### LINKED ISSUE #5742: Move DevMenu to XAML markup file
We currently have a few places where we have a large-ish control that gets created by parsing a string of text with xaml markup at runtime. This makes it difficult to use something like the designer and live visual tree to iterate or debug issues.

RedBox and DevMenu are two such controls that follow this pattern

### PR DESCRIPTION
**Pull request by asklar**
Fixes #5742 

This one is a doozy. I had set out to get rid of the "XamlReader parsing a string" pattern so as to make it easier for us to iterate on the visuals and make debugging easier. In trying to get there I found a number of bugs and learned some things.

1. [cppwinrt doesn't sanitize the namespace name for its controls](https://github.com/microsoft/cppwinrt/issues/728). For the `Microsoft.ReactNative` project, the root namespace is naturally `Microsoft.ReactNative`, and this is used by winmd and other tools. The cppwinrt VSIX has a bug where it will use the `RootNamespace` as the namespace to plop new user controls/pages/etc. This works ok as long as the name is a valid C++ name. In our case, we have a period so this means cppwinrt generates invalid (non-building) C++ code.

2. [cppwinrt user controls/pages cannot be created in project subfolders, only at the root](https://github.com/microsoft/cppwinrt/issues/731). The code that cppwinrt generates does not take into account the root-relative folder in which a control or page is created. It expects that the generated headers will be at the root - or that the folder be available via `AdditionalIncludeDirectories`. WinUI 2 gets around this by adding each control's directory to `AdditionalIncludeDirectories` which is less than ideal.

3. [XAML generates type metadata for Bindable properties regardless of whether any markup references them or not](https://github.com/microsoft/microsoft-ui-xaml/issues/3133). This means that the first time one adds markup to a project, if the project references MUXC (WinUI 2) then the compiler needs to be able to find any and all XAML types marked with Bindable - such as `TreeViewNode`, regardless of whether the `TreeViewNode` type is used at all.

Moving the control to XAML makes the code in DevMenu cleaner (no looking up an element by name, instead we use property accessors). Ideally we would use bindings to set the text blocks' text but this is ok for now.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5744)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Pch/pch.h [hunk 1]
```diff
@@ -53,3 +53,7 @@
 
 #include <activeObject/activeObject.h>
 #include <future/future.h>
+
+// https://github.com/microsoft/microsoft-ui-xaml/issues/3133
+#include <winrt/Microsoft.UI.Xaml.Controls.h>
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Pch/pch.h:58]: Does this need to be changed to wrt WinUI3?

Comment (asklar) [reply vnext/Microsoft.ReactNative/Pch/pch.h:58]: @jonthysell good question, it does not :) Bug 3133 is that if you reference winui 2.x then xaml compiler will try to generate type info in your app for any type marked Bindable which includes TreeViewNode, even if you never use TreeViewNode.
The same header exists in 3.0 so no change necessary.

**Comment by asklar** [comment]
> Microsoft.UI.Xaml.Controls (or at least the WINUI 3 version) adds something like 300MB to our PCH. It was removed along with some other headers since it was a large contributor to us and customers seeing compiler OOM issues. Can we avoid adding such a large header back to the PCH? Not sure if the winui2 one has the same impact, but think we need to measure since we're already quite sensitive here, and this would put winui 3 builds in a worse state.

Thanks for bringing it up; The WinUI 2.x version of the muxc header adds about 50 MB
- Without MUXC (playground, massaged to ensure we don't transitively include MUXC): 1184038904 
- The above + MUXC: 1233977336
This gives a difference of 47.6 MB. It's not great, but it's also not 300 MB. The guidance on cppwinrt is starting to shift towards trying to reduce usage of PCHs from what I've heard, anyway, so this might not even be relevant in a few months.

The reason we have to bring this header in, is because of a xaml compiler bug that will likely not be fixed in the OS: https://github.com/microsoft/microsoft-ui-xaml/issues/3133 but it will be fixed in winui 3 instead. Therefore this becomes a tradeoff between paying the 47 MB of PCH and getting XAML markup and its inner loop (designer, compile time checking for correctness, etc.), so I still feel it is worthwhile.

**Comment by NickGerleman** [comment]
@asklar 50MB for non-WinUI3 builds isn't too bad. A more concrete concern was whether this would worsen the WinUI 3 CI, since we include a header that is IIRC much larger on it. The OOM issues we're seeing with WinUI3 are during normal compilation instead of creating the PCH, but IIRC MSVC will reserve virtual memory for the entire PCH for compiling every translation unit.

The builds in CI have been impacting productivity of folks since the WinUI version bump lead to greater instability. If we're doing something that has the potential to make it worse for folks again, I think we would need to take some steps to try to mitigate the issue. E.g. making WinUI 3 jobs warn instead of fail CI, removing parallelism, etc. I was hoping this would happen more shortly after the WinUI specific builds started causing issues for developers, but the impact to engineers across RNW has been pretty severe.

**Pull request merged**